src/jit/codegenlegacy.cpp

```diff
@@ -11371,8 +11371,6 @@ void CodeGen::genCodeForTreeSmpOpAsg(GenTreePtr tree)
 
             if (VarSetOps::IsMember(compiler, genUpdateLiveSetForward(op2), varDsc->lvVarIndex))
             {
-                noway_assert(compiler->opts.compDbgCode);
```

Comment:
Ok. Then I will create an issue for it and revert the change since I don't think this has something to do with inline fix here.